client/app/scripts/components/node-details.js

```diff
@@ -77,14 +77,19 @@ const NodeDetails = React.createClass({
     const details = this.props.details;
     const nodeExists = this.props.nodes && this.props.nodes.has(this.props.nodeId);
 
+    if (details) {
+      return this.renderDetails();
+    }
+
```

Comment:
A 404 on getting node details will result in an empty `details` field, so that will work. Maybe rename `nodeExists` to `nodeInTopology` because it may be filtered. But to assuage paul's concerns, all fields rendered from the `details` object so they dont rely on data from the topology endpoint.